server/entity/falling_block.go

```diff
@@ -57,13 +90,37 @@ func (f *FallingBlock) Tick(w *world.World, _ int64) {
 	f.mu.Unlock()
 
 	m.Send()
-	pos := cube.PosFromVec3(m.pos)
 
+	distThisTick := f.vel.Y()
+	if distThisTick < f.fallDistance.Load() {
+		f.fallDistance.Sub(distThisTick)
+	} else {
+		f.fallDistance.Store(0)
+	}
```

Comment:
I have some difficulty making sense of what happens here. I assume the fall distance is supposed to be reset if the falling block has upwards velocity, but what exactly is the `distThisTick < f.fallDistance.Load()` check supposed to do in terms of behaviour?

Reply:
It's the same behaviour copied from the player package.

Reply:
Are you able to make sense of it? Do we have any kind of reference to what this is based on?

Reply:
See https://github.com/df-mc/dragonfly/blob/17292335c97dc66c42b4b28fba7ea08a5e105b28/server/player/player.go#L476. When the block is falling, the velocity will be negative which will be under the fall distance since it starts at zero. Otherwise, the block isn't falling so the fall distance should be zero.

Reply:
I get that part, but when the fall distance is, say, 60 blocks, you would need a velocity of 60 upwards to cancel out the fall damage if I understand this correctly, which... doesn't seem sensible?

Reply:
Resolving after discussion on Discord.